nomad/job_endpoint_validators.go

```diff
@@ -0,0 +1,49 @@
+package nomad
+
+import (
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/pkg/errors"
+)
+
+type jobNamespaceConstraintCheckHook struct {
+	srv *Server
+}
+
+func (jobNamespaceConstraintCheckHook) Name() string {
+	return "namespace-constraint-check"
+}
+
+func (c jobNamespaceConstraintCheckHook) Validate(job *structs.Job) (warnings []error, err error) {
+	// This was validated before and matches the WriteRequest namespace
+	ns, err := c.srv.State().NamespaceByName(nil, job.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: Should this not get checked by another validator before?
```

Comment:
Probably. But it's a very cheap extra check so having it here lets us freely reorder hooks without worry and prevents us from accidentally crashing unit tests by passing in bad data 😁